summarization/extractor_output_preprocessor.py

```diff
@@ -17,12 +17,25 @@ class SentenceWithAttributes:
     embedding or font_style
     '''
 
-    def __init__(self, text, index, font_style, embedding):
+    def __init__(
+            self,
+            text,
+            para_index,
+            sentence_index_in_para,
+            sentence_weight,
+            font_style,
+            embedding):
         self.text = text
-        self.index = index
+        self.paragraph_index = para_index
+        self.sentence_index_in_para = sentence_index_in_para
+        self.sentence_weight = sentence_weight
```

Comment:
What are sentence_weights and why do we need them?

Reply:
the sentence_weight is 1 / (number of sentences in the para which the sentence belongs to ) 
so previously we store a floating point index,
now we separate it and store it so we can recombine it whenever necessary as :
para_index + sentence_index_in_para * sentence_weight